src/psyclone/psyGen.py

```diff
@@ -1042,13 +1044,28 @@ class Directive(Node):
     # The prefix to use when constructing this directive in Fortran
     # (e.g. "OMP"). Must be set by sub-class.
     _PREFIX = ""
+    # Class attributes of this node textual representations
+    _children_valid_format = "Schedule"
+    _text_name = "Directive"
+    _colour_key = "Directive"
 
     def __init__(self, ast=None, children=None, parent=None):
         # A Directive always contains a Schedule
         sched = self._insert_schedule(children, ast)
         super(Directive, self).__init__(ast, children=[sched], parent=parent)
-        self._text_name = "Directive"
-        self._colour_key = "Directive"
+
+    @staticmethod
+    def _validate_child(position, child):
+        '''
+        :param int position: a position to be validated.
+        :param child: a child to be validated.
+        :type child: :py:class:`psyclone.psyir.nodes.node`
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        '''
+        return position == 0 and isinstance(child, Schedule)
```

Comment:
I see your point and it is a nice idea. I think we don't even need the dictionary as python eval may already be able convert the strings to the class type. But since getting a regex parser right may be tricky I think it is better to leave this to a separate PR.

Also, forcing the validation to be always done with a true regex expression could limit the PSyIR language in the future. Are we sure a Regular Grammar will cover all node types and hierarchies we want to add?

Reply:
I'm certainly not sure :-). If you think this is something worth looking into then could you create a new issue and add a reference to it in the code. If not, don't worry, it was just a thought.

Reply:
I created an issue and added references to it.